source/extensions/common/aws/credentials_provider_impl.cc

```diff
@@ -435,14 +436,29 @@ TaskRoleCredentialsProvider::TaskRoleCredentialsProvider(
           envoy::config::cluster::v3::Cluster::STATIC /*cluster_type*/, credential_uri),
       credential_uri_(credential_uri), authorization_token_(authorization_token) {}
 
-bool TaskRoleCredentialsProvider::needsRefresh() {
+bool ContainerCredentialsProvider::needsRefresh() {
   const auto now = api_.timeSource().systemTime();
   return (now - last_updated_ > REFRESH_INTERVAL) ||
          (expiration_time_ - now < REFRESH_GRACE_PERIOD);
 }
 
-void TaskRoleCredentialsProvider::refresh() {
-  ENVOY_LOG(debug, "Getting AWS credentials from the task role at URI: {}", credential_uri_);
+void ContainerCredentialsProvider::refresh() {
+  ENVOY_LOG(debug, "Getting AWS credentials from the container role at URI: {}", credential_uri_);
+
+  // ECS Task role: use const authorization_token set during initialization
+  absl::string_view authorization_header = authorization_token_;
+  absl::StatusOr<std::string> token_or_error;
+
+  if (authorization_token_.empty()) {
+    // EKS Pod Identity token is sourced from AWS_CONTAINER_AUTHORIZATION_TOKEN_FILE
+    if (const auto token_file = std::getenv(AWS_CONTAINER_AUTHORIZATION_TOKEN_FILE)) {
```

Comment:
@suniltheta Per comment on const - authorization_token_ now remains const (ECS use case), however if it is not populated (EKS Pod Identity use case) then it will attempt to read from file instead. If this succeeds, then authorization_header will now contain the file token and will be used. So original const is honoured.